src/blockchain_db/sqlite/db_sqlite.cpp

```diff
@@ -1102,4 +1144,29 @@ bool BlockchainSQLite::save_payments(
     }
     return true;
 }
+
+void BlockchainSQLite::set_rewards_hf21(const std::unordered_map<eth::address, uint64_t>& rewards) {
+    log::trace(
+            logcat, "BlockchainSQLite::{} removing OXEN rewards and adding SENT rewards", __func__);
+
+    // values in `rewards` represent converted OXEN -> SENT rewards,
+    // any unconverted are dropped (but were paid out last block anyway).
+    db.exec("DELETE FROM batched_payments_accrued");
+
+    auto insert_payment = prepared_st(
+            "INSERT INTO batched_payments_accrued (address, payout_offset, amount) VALUES (?, ?, "
+            "?)");
+
+    for (auto& [addr, amt] : rewards) {
+        auto address_str = "0x{:x}"_format(addr);
+        auto amount = static_cast<int64_t>(amt * BATCH_REWARD_FACTOR);
```

Comment:
Should be covered in #1802 but github won't let me comment here in that PR.  I assume that this `amt * BATCH_REWARD_FACTOR` should be converted to the "db vs actual" coin functions?

Reply:
Yep. Just separated them because that change could be completely isolated out and that this PR focuses on just getting mocknet in.